azext_edge/tests/edge/checks/base/test_deployment_checks_unit.py

```
@@ -0,0 +1,266 @@
+                        }
+                    },
+                },
+                {
+                    "description": "Evaluate cluster nodes",
+                    "name": "evalClusterNodes",
+                    "status": "success",
+                    "targets": {
+                        "cluster/nodes": {
+                            "_all_": {
+                                "conditions": ["len(cluster/nodes)>=1"],
+                                "evaluations": [{"status": "success", "value": {"len(cluster/nodes)": 1}}],
+                                "status": "success",
+                            }
+                        },
+                        "cluster/nodes/k3d-k3s-default-server-0": {
+                            "_all_": {
+                                "conditions": [
+                                    "info.architecture in (amd64)",
+                                    "allocatable.cpu>=4",
+                                    "allocatable.memory>=16G",
+                                ],
+                                "evaluations": [
+                                    {
+                                        "status": "error",
+                                        "value": {"info.architecture": "invalid_arch"},
+                                    },  # invalid architecture
+                                    {"status": "error", "value": {"allocatable.cpu": 1}},  # invalid cpu count
+                                    {"status": "success", "value": {"allocatable.memory": "12G"}},  # invalid memory
+                                ],
+                                "status": "success",
+                            }
+                        },
+                    },
+                },
+            ],
+            True,
+        ),
+    ],
+    ids=["all_success", "errors"],
+)
+@pytest.mark.parametrize(
+    "acs_config",
+    [
+        {"feature.diskStorageClass": "default,local-path"},
```

What about a case for `{"feature.diskStorageClass": ""}` ?

---

I see https://github.com/Azure/azure-iot-ops-cli-extension/blob/ea2f9a3da723fe8e6312876874982250ad90471a/azext_edge/tests/edge/checks/base/test_deployment_checks_unit.py#L47 has a case for singleclass

---

This brings up an interesting question - What is the expected behavior if a customer overrides the storageclass property with an empty value?

`az iot ops init -g test --cluster test --check-cluster --acs-config feature.diskStorageClass="" `

Right now the result is an error, but the condition looks odd:

```
Cluster readiness pre-checks failed:
        Target 'cluster/storage-classes':
                Expected: 'contains(cluster/storage-classes, any())', Actual: 'local-path'
```

Should we throw a validation error if they're trying to `check` an init without a defined storageclass in the config? Or should we pass / continue since we don't know what to check for and therefore... anything is considered a success?

My guess is that this might break some config down the line and therefore we should error.

---

does this make more sense to raise as a `ValidationError` in the `cluster_kwarg` building, or should we have that validation fire in the `_check_storage_classes` function?

Or should we just output the standard pre-check error, rather than raising a `ValidationError`?

---

Let's validate service behavior both when there is an empty or null value for feature.diskStorageClass. Want to make sure it doesn't default to something.

In the case it doesn't default (and --enable-fault-tolerance is not True), user theoretically shouldn't need to use --check-cluster for a validation error since we presume it will error without one.